addons/mail/static/src/core/common/store_service.js

```diff
@@ -583,12 +582,15 @@ export class Store extends BaseStore {
     }
 
     async joinChat(id, forceOpen = false) {
-        const data = await rpc("/discuss/channel/get_or_create_chat", {
+        const { channel_id, data } = await rpc("/discuss/channel/get_or_create_chat", {
             partners_to: [id],
-            force_open: forceOpen,
         });
-        const { Thread } = this.store.insert(data);
-        return Thread[0];
+        this.store.insert(data);
+        const thread = this.store.Thread.get({ id: channel_id, model: "discuss.channel" });
+        if (forceOpen) {
+            thread.openChatWindow();
+        }
+        return thread;
```

Comment:
`store.joinChat()` is a general-purpose function to join a dm/group chat. It shouldn't be biaised and open chat window if the view has no chat hub visible like in discuss app.

Reply:
I didn't change this behavior, that's probably why `forceOpen` is a param with default as `false`.

Reply:
Looks like dead code, probably since the introduction of ctrl-K (and removal of "new message" chat window)

Reply:
What do you mean dead code? The function is called with the param with either true or false in different places.

Reply:
Sorry, my IDE was lying, indeed this is still useful